### PR TITLE
Add a warning when there are extensions incompatible with WooPay

### LINKED ISSUE #6422: Add a warning if merchant has extensions incompatible with WooPay
### Description
WooPay has some incompatibilities with some known plugins. We need to add a warning to the settings page so people know that they have some known extensions that are know to be incompatible with WooPay. 

### Acceptance criteria
- The list on known incompatible extensions should be updatable at any time
- Changes should reflect at the store within 24 hours
- WooPay should not be disabled automatically if we find incompatible extensions

The message in the settings page should be:
"WooPay has been disabled because one or more of your extensions are incompatible. Learn more"

### Designs
image

Figma file: C1r0Vzxh1mhbDJJnbjHp5q-fi-2126_77381

### Additional context
Slack thread: p1685485221526939-slack-C02CCT3F1QA
P2: pdpOw2-2TC-p2

### PR DESCRIPTION
**Pull request by leonardola**
#### Changes proposed in this Pull Request
This PR introduces e warning for merchants when incompatible plugins are installed. This will not prevent merchants from activating WooPay. The list of incompatible extensions is updated once a day.

Fixes #6422

#### Testing instructions
- Pull the latest changes from WooPay
- Checkout to this branch in your merchant site.
- Go to the WooPay \wp-admin > WooPay and add a plugin that you have installed in in your merchant site to the list of incompatible plugins (It's just the plugin name like woocommerce-payments not woocommerce-payments/woocommerce-payments.php )
- Save
- Go to your merchant site
- Install a plugin that allows you to manage cron jobs manually
- Execute the `validate_incompatible_extensions` cron
- Go to \wp-admin > Payments > Settings > Express checkout
- Make sure the message below shows up

Screenshot 2023-06-09 at 11 03 25 AM

- Remove the extension from the incompatible ones and run the cron again. The warning should not show up anymore

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6483` or branch name `add/6422-warning-for-woopay-incompatible-extensions` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/6422-warning-for-woopay-incompatible-extensions) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 257dbb0d7cb25eee18107804ec35baefed515adb
- Build time: 2023-06-13 15:08:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-blocks-payment-method.php [hunk 1]
```diff
@@ -85,7 +86,7 @@ public function get_payment_method_data() {
 
 		if ( $is_woopay_eligible && $is_woopay_enabled ) {
 			$woopay_config = [
-				'woopayHost' => defined( 'PLATFORM_CHECKOUT_FRONTEND_HOST' ) ? PLATFORM_CHECKOUT_FRONTEND_HOST : 'https://pay.woo.com',
+				'woopayHost' => WooPay_Utilities::get_woopay_url(),
```
Comment (malithsen) [inline includes/class-wc-payments-blocks-payment-method.php:89]: Kudos for refactoring this

**Review by malithsen — APPROVED** [review]
Thanks for the PR @leonardola 
Code looks good to me and tests well. Approving 🚀 

I had the following two questions while reviewing but they are not blockers.

1. If a merchant disables the incompatible extension, the notice will not immediately go away. Instead, it will only clear out after the next run of the scheduled event. Is this expected behavior, or should we recheck immediately after a plugin deactivation?
2. Are we still planning on auto-disabling WooPay when an incompatibility is found? If that's not the case, contrary to my previous comment in https://wp.me/pdpOw2-2TC#comment-3122, I'd suggest adding a single event `wcadmin_woopay_extension_incompatibility_notice_shown` to track the instances where the notice is shown, this can be a separate PR. cc @pierorocca

**Comment by pierorocca** [comment]
> 2\. Are we still planning on auto-disabling WooPay when an incompatibility is found?

No longer the case. We'll notify only and not auto disable.

**Comment by pierorocca** [comment]
> 1. If a merchant disables the incompatible extension, the notice will not immediately go away. Instead, it will only clear out after the next run of the scheduled event. Is this expected behavior, or should we recheck immediately after a plugin deactivation?

Either is fine. If real-time or event based is possible, that's preferred.

**Comment by pierorocca** [comment]
> Either is fine. If real-time or event based is possible, that's preferred.

I should add though on install of WCPay, the check should happen then and not wait for a cron job to display the notification 24 hours later. By that time the merchant might be in production. @leonardola will the detected incompatibility be available at install?

**Comment by leonardola** [comment]
@pierorocca The scheduler runs right away when the plugin is installed and then once a day. It won't be triggered right away for new extensions though. Do we need that?

**Comment by pierorocca** [comment]
If possible, having the job trigger on any plugin/extension install I think would make more sense than only when WooPayments is installed. Is that a big modification and something for a future optimization or a small, quick update?

**Comment by leonardola** [comment]
That should be a small change. We just need to use the [activated_plugin](https://developer.wordpress.org/reference/hooks/activated_plugin/) hook

**Pull request merged**